### PR TITLE
Add TLF Support to the Client Side

### PR DESCRIPTION
**Pull request by jxguan**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -112,35 +118,60 @@ func createClientWithClient(ctx context.Context, searchCli sserver1.SearchServer
...
-	relPath, err := relPathStrict(c.directory, pathname)
+// AddFile indexes a file in `directory` with the given `pathname` and writes
+// the index to the server.
+func (c *Client) AddFile(directory, pathname string) error {
+	absDir, err := filepath.Abs(directory)
+	if err != nil {
+		return err
+	}
+
+	if _, ok := c.directoryInfos[absDir]; !ok {
```
Comment (strib) [inline client/client.go]: `directoryInfos` needs to be protected with a lock, right?  Client should be goroutine-safe, I think.

File: client/client.go [hunk 2]
```diff
@@ -112,35 +118,60 @@ func createClientWithClient(ctx context.Context, searchCli sserver1.SearchServer
...
+		return errors.New("invalid directory name provided")
+	}
+
+	relPath, err := relPathStrict(absDir, pathname)
 	if err != nil {
 		return err
 	}
 
-	docID, err := pathnameToDocID(relPath, c.pathnameKey)
+	docID, err := pathnameToDocID(relPath, c.directoryInfos[absDir].pathnameKey)
```
Comment (strib) [inline client/client.go]: Might as well save `c.directoryInfos[absDir]` in a local variable so you don't have to look it up so much below.

File: client/client/main.go [hunk 3]
```diff
@@ -40,91 +40,100 @@ func addAllFiles(cli *client.Client, lastIndexed time.Time) filepath.WalkFunc {
...
-	filenames, err := cli.SearchWordStrict(keyword)
-	if err != nil {
-		fmt.Printf("Error when searching word %s: %s", keyword, err)
-		return
+// performSearchWord searches for the word `keyword` on `cli`, and prints out
+// the results.
+func performSearchWord(cli *client.Client, clientDirs []string, keyword string) {
+	var allFiles []string
+	for _, clientDir := range clientDirs {
+		filenames, err := cli.SearchWordStrict(clientDir, keyword)
```
Comment (strib) [inline client/client/main.go:126]: Maybe add a TODO here that we can parallelize these calls for better performance.

File: client/client.go (hunk 1 above)
Comment (strib) [reply client/client.go]: Actually I guess it's readonly.  So no need for locking for now.  Later we might want to add directories during runtime though, and then we'll need a lock.

**Comment by strib** [comment]
Looks good mod comments, thanks!

File: client/client.go (hunk 1 above)
Comment (jxguan) [reply client/client.go]: K I'll add a TODO

File: client/client.go (hunk 2 above)
Comment (jxguan) [reply client/client.go]: Fixed

File: client/client/main.go (hunk 3 above)
Comment (jxguan) [reply client/client/main.go:126]: Fixed

**Pull request merged**